crates/stc_ts_file_analyzer/src/analyzer/convert/mod.rs

```diff
@@ -225,7 +224,34 @@ impl Analyzer<'_, '_> {
                         metadata: Default::default(),
                     }),
 
-                    _ => d.type_ann.validate_with(child)?,
+                    t => {
+                        if let RTsType::TsTypeRef(type_ref) = t {
+                            if let RTsEntityName::Ident(t) = &type_ref.type_name {
+                                match &*t.sym {
+                                    "Uppercase" | "Lowercase" | "Capitalize" | "Uncapitalize" => {
```

Comment:
String intrinsic types in type alias assignments aren't caught by the https://github.com/dudykr/stc/blob/7bfe8ea0a4b4407f6517c8978ecd3d04de7afdb8/crates/stc_ts_file_analyzer/src/analyzer/convert/mod.rs#L199-L201
 match arm so thats why we're doing this